plugin/meshnet.go

```diff
@@ -75,20 +76,20 @@ func loadConf(bytes []byte) (*netConf, *current.Result, error) {
 	return n, result, nil
 }
 
-// getVxlanSource uses netlink to query the IP and LinkName of the interface with default route.
-func getVxlanSource() (string, string, error) {
-	log.Infof("Looking up a default route to get the intf and IP for vxlan")
-	r, err := netlink.RouteGet(net.IPv4(1, 1, 1, 1))
-	if (err != nil) && len(r) < 1 {
-		return "", "", fmt.Errorf("failed to get default route: %s\n%+v", err, r)
-	}
-	srcIP := r[0].Src.String()
-	link, err := netlink.LinkByIndex(r[0].LinkIndex)
-	if err != nil {
-		return "", "", fmt.Errorf("failed to get link by its index: %s", err)
+// getVxlanSource uses netlink to get the iface reliably given an IP address.
+func getVxlanSource(nodeIP string) (string, string, error) {
+
```

Comment:
Can you add a separate error check for `nodeIP == ""`? e.g. if someone drops the `HOST_IP` env var accidentally, this would fall through to the return statement of this function and the error would be "No iface found for address ", which is not very informative.